### ISSUE TITLE
YouTube Support

### ISSUE THREAD

**Issue by Zibbp** [Zibbp/ganymede#16]

**Comment by lpuv** [comment]
Would it be possible to implement this without chat support/emote support for now?

**Comment by Zibbp** [comment]
Hi,
No plans to add any form of YouTube support in the near future. In the meantime you can manually download a YouTube stream/video using `yt-dlp` and manually import it into Ganymede in the UI or by using the API.

**Comment by Zibbp** [comment]
Authentication only supports cookies so you'll either need to use the cookie in your browser or perform an auth request in your python script. 
```
POST http://ip:port/api/v1/auth/login
```
with the following body
```json
{
  "username": "my_username",
  "password": "my_password"
}
```
Then you need to use the cookie returned by the response (`access-token`) for any future requests. Not sure how to do this in Python, but probably easy with `requests`.

For the request body for creating a VOD it looks like you have it mostly right. I would reference the [http/vod.go CreateVodRequest struct](https://github.com/Zibbp/ganymede/blob/main/internal/transport/http/vod.go#L40) for details of what is required.

It may also be useful to perform a `GET /api/v1/vod` to see what fields are populated for your current vods.

Let me know if there's anything else I can help with. I don't like to advertise the API much as I tend to change it between versions.

Related, are you downloading chats for your YouTube vods or just the video/stream?

**Comment by drag0n141** [comment]
Is it now possible with the newest change to yt-dlp to download now youtube live streams?

**Comment by Zibbp** [comment]
No, no other platforms are supported at this time. Maybe someday additional platforms will be supported.

**Comment by BigFlubba** [comment]
So I just discovered [EOS](https://github.com/Zibbp/eos). Is it possible that the functions of EOS for existing libraries can be combined into Ganymede or EOS is more updated to reflect more of Ganymede? I don't care about downloading or anything like that. I already have [Tubeless (Pinchflat fork)](https://github.com/CommunityMaintained/tubeless/) to handle the downloading and management. I just need a front-end to serve them. Jellyfin (with plugins) does work but Jellyfin is not designed for this.
